src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -352,9 +353,13 @@ public static boolean checkTree(@NotNull final IWorldReader world, final BlockPo
         }
 
         // Only harvest nearly fully grown dynamic trees(8 max)
+        int treeSize = 0;
+        if(building != null) {
```

Comment:
formatting. And why is the building nullable?

Reply:
To account for problems when the forester gets fired while trying to path to a tree (his worker hut probably isn’t a BuildingLumberjack anymore in that case)

Reply:
Looking through the code once again, I found a better way to get the setting over here